portable/NetworkInterface/LPC54018/NetworkInterface.c

```diff
@@ -129,197 +131,205 @@ void ENET_IntCallback( ENET_Type * base,
+				}
+			}
+		}
+	}
 }
 
 BaseType_t xGetPhyLinkStatus( void )
 {
-    bool link;
-
-    PHY_GetLinkStatus( &phyHandle, &link );
-    return link ? pdTRUE : pdFALSE;
+	return g_linkStatus ? pdTRUE : pdFALSE;
 }
 
 
 BaseType_t xNetworkInterfaceInitialise( void )
 {
-    enet_config_t config;
-    uint32_t refClock = 50000000; /* 50MHZ for rmii reference clock. */
-    phy_speed_t speed;
-    phy_duplex_t duplex;
-    status_t status;
-    bool link = false;
-
-    phy_config_t phyConfig;
-
-    int bufferIndex;
-
-    for( bufferIndex = 0; bufferIndex < ENET_RXBD_NUM; bufferIndex++ )
-    {
-        rxbuffer[ bufferIndex ] = ( uint32_t ) &receiveBuffer[ bufferIndex ];
-    }
-
-    phyConfig.phyAddr = PHY_ADDRESS;
-    phyConfig.autoNeg = true;
-    mdioHandle.resource.base = ENET;
-
-    /* prepare the buffer configuration. */
-    enet_buffer_config_t buffConfig[ 1 ] =
-    {
-        {
-            ENET_RXBD_NUM, ENET_TXBD_NUM,
-            &g_txBuffDescrip[ 0 ], &g_txBuffDescrip[ 0 ],
-            &g_rxBuffDescrip[ 0 ], &g_rxBuffDescrip[ ENET_RXBD_NUM ],
-            &rxbuffer[ 0 ], ENET_BuffSizeAlign( ENET_RXBUFF_SIZE ),
-        }
-    };
-
-    while( !link )
-    {
-        status = PHY_Init( &phyHandle, &phyConfig );
-
-        if( kStatus_Success == status )
-        {
-            PHY_GetLinkStatus( &phyHandle, &link );
-        }
-        else if( kStatus_PHY_AutoNegotiateFail == status )
-        {
-            PRINTF( "\nPHY Auto-negotiation failed. Please check the cable connection and link partner setting.\n" );
-        }
-        else
-        {
-            PRINTF( "\nUnknown PHY failure %d\n", status );
-        }
-    }
-
-    PHY_GetLinkSpeedDuplex( &phyHandle, &speed, &duplex );
-
-    /* Get default configuration 100M RMII. */
-    ENET_GetDefaultConfig( &config );
-
-    /* Use the actual speed and duplex when phy success to finish the autonegotiation. */
-    config.miiSpeed = ( enet_mii_speed_t ) speed;
-    config.miiDuplex = ( enet_mii_duplex_t ) duplex;
-
-    /* Initialize ENET. */
-    ENET_Init( ENET, &config, g_macAddr, refClock );
-
-    /* Enable the rx interrupt. */
-    ENET_EnableInterrupts( ENET, ( kENET_DmaRx ) );
-
-    /* Initialize Descriptor. */
-    ENET_DescriptorInit( ENET, &config, &buffConfig[ 0 ] );
-
-    /* Create the handler. */
-    ENET_CreateHandler( ENET, &g_handle, &config, &buffConfig[ 0 ], ENET_IntCallback, NULL );
-    NVIC_SetPriority( 65 - 16, 4 ); /* TODO this feels like a hack and I would expect a nice ENET API for priority. */
-
-    /* Active TX/RX. */
-    ENET_StartRxTx( ENET, 1, 1 );
-
-    if( xTaskCreate( rx_task, "rx_task", 512, NULL, ( configMAX_PRIORITIES - 1 ), &receiveTaskHandle ) != pdPASS )
-    {
-        PRINTF( "Network Receive Task creation failed!.\n" );
-
-        while( 1 )
-        {
-        }
-    }
-
-    return pdTRUE;
+BaseType_t returnValue = pdFALSE;
+static enum {initPhy, startReceiver, waitForLink, configurePhy }networkInitialisePhase = initPhy;
```

Comment:
The switch is a little confusing because it uses "fall through" without a warning.

I would prefer to see some comment:
~~~c
                }
            }
            /* Fall through. */
        case startReceiver:
            networkInitialisePhase = startReceiver;
~~~

Reply:
added comments